modules/profile/manifests/st2server.pp

```diff
@@ -1,4 +1,286 @@
+    ssl               => true,
+    ssl_cert          => $_ssl_cert,
+    ssl_key           => $_ssl_key,
+    ssl_protocols     => $_ssl_protocols,
+    ssl_ciphers       => $_cipher_list,
+    vhost_cfg_prepend => $_ssl_options,
+    server_name       => $_server_names,
+    add_header        => $_headers,
+    www_root          => '/opt/stackstorm/static/webui/',
+    require           => X509_cert[$_ssl_cert],
+  }
+
+  ## st2auth and st2api SSL proxies via nginx
+
+  ### Shared proxy headers for each reverse proxy
+  nginx::resource::vhost { 'st2api':
+    ensure               => present,
+    listen_ip            => $_host_ip,
+    listen_port          => 9101,
+    ssl                  => true,
+    ssl_port             => 9101,
+    ssl_cert             => $_ssl_cert,
+    ssl_key              => $_ssl_key,
+    ssl_protocols        => $_ssl_protocols,
+    ssl_ciphers          => $_cipher_list,
+    server_name          => $_server_names,
+    vhost_cfg_prepend    => $_ssl_options,
+    proxy                => 'http://st2api',
+  }
+
+  nginx::resource::upstream { 'st2api':
+    members => ["127.0.0.1:${_st2api_port}"],
+  }
+
+  # ## Authentication
+  # ### Nginx needs access to make calls to PAM, and by
+  # ### extension, needs access to /etc/shadow to validate users.
+  # ### Let's at least try to do this safely and consistently
+
+  # Let's add our nginx user to the `shadow` group, but do
+  # it after the package manager has installed and setup
+  # the user
+  user { $_nginx_daemon_user:
+    groups  => ['shadow'],
+  }
+
+  pam::service { 'nginx':
+    content => '@include common-auth',
+  }
+
+  nginx::resource::vhost { 'st2auth':
+    ensure               => present,
+    listen_ip            => $_host_ip,
+    listen_port          => 9100,
+    ssl                  => true,
+    ssl_port             => 9100,
+    ssl_cert             => $_ssl_cert,
+    ssl_key              => $_ssl_key,
+    ssl_protocols        => $_ssl_protocols,
+    ssl_ciphers          => $_cipher_list,
+    vhost_cfg_prepend    => $_ssl_options,
+    server_name          => $_server_names,
+    location_raw_prepend => [
+      'auth_pam "Restricted";',
+      'auth_pam_service_name "nginx";',
+    ],
+    proxy                => 'http://st2auth',
+  }
+
+  nginx::resource::upstream { 'st2auth':
+    members => ["127.0.0.1:${_st2auth_port}"],
+  }
+
+  # Setup the installer on initial provision, and get rid of it
+  # after setup has been run.
+  if ! $_installed {
+    vcsrepo { '/opt/stackstorm/st2installer':
+      ensure   => present,
+      provider => 'git',
+      source   => 'https://github.com/stackstorm/st2installer',
+      require  => Class['::st2::profile::server'],
+    }
+
+    uwsgi::app { 'st2installer':
+      ensure              => present,
+      uid                 => $_nginx_daemon_user,
+      gid                 => $_nginx_daemon_user,
+      application_options => {
+        'http-socket'  => "127.0.0.1:${_st2installer_port}",
```

Comment:
This is how I was using uwsgi with mistral api:
`uwsgi -s /opt/stackstorm/mistral_api.sock --wsgi-file <MISTRAL>/mistral/api/wsgi.py -H /opt/openstack/mistral/.venv/ `

nginx config:
https://gist.githubusercontent.com/DoriftoShoes/d3ae68f0b3a9ce77a163/raw/feb7d57dcf88dc47210b02619e2ae5129447621c/mistral.conf